### ISSUE TITLE
Unable to install dotnet-ef 3.1-preview1

### ISSUE THREAD

**Issue by MintPlayer** [aspnet/EntityFrameworkCore#18723]
I'm trying to use the **dotnet ef** tool in my database project for ASP.NET Core.

Installed NuGet packages (project):

- Microsoft.EntityFrameworkCore.SqlServer @ 3.1.0-preview1.19506.2
- Microsoft.EntityFrameworkCore.Tools @ 3.1.0-preview1.19506.2

When I run a dotnet ef command in my project, I'm getting the following error:

```
C:\WINDOWS\system32>dotnet --version
3.1.100-preview1-014459

C:\WINDOWS\system32>dotnet ef --version
Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET Core program, but dotnet-ef does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
```

Okay, dotnet-ef is not installed. So let's install it globally (admin cmd):

```
C:\WINDOWS\system32>dotnet tool install dotnet-ef --global --version="3.1.0-preview1.19506.2"
error NU1202: Package dotnet-ef 3.1.0-preview1.19506.2 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1) / any. Package dotnet-ef 3.1.0-preview1.19506.2 supports: netcoreapp3.1 (.NETCoreApp,Version=v3.1)
The tool package could not be restored.
Tool 'dotnet-ef' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET Core tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
```

So the response is basically telling me that the tool is not compatible with netcoreapp3.1 and only supports netcoreapp3.1, which doesn't make any sense.

- Version specified -> check
- dotnet-ef is a .NET Core tool
- The `%userprofile%\.dotnet\tools folder is definitely on the path variable (User scope). Didn't have to change anything after installation

I picked preview1 since I cannot yet install the preview2 runtime/sdk on my computer.

### Further technical details

**EF Core version:** 3.1.0-preview1.19506.2
**Database provider:** Microsoft.EntityFrameworkCore.SqlServer
**Target framework:** .NET Core 3.1.0-preview1 (THE ONLY ONE I installed on my fresh installed Windows system, but anyway below a list of all installed runtimes)
**Operating system:** Windows 10 Home 1903
**IDE:** Visual Studio 2019 16.4.0 Preview 2.0 (Again no other VS installed)

### Installed runtimes:

```
C:\WINDOWS\system32>dotnet --list-runtimes
Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.0-preview1.19508.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.0-preview1.19506.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.1.0-preview1.19506.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

Is this a bug? Additional info or repository needed?

**Comment by ajcvickers** [comment]
@MintPlayer Can you try this with preview 2, which is now on NuGet? We believe the issue has already been fixed.

**Comment by bricelam** [comment]
Fixed by #18551

**Comment by MintPlayer** [comment]
Done.

1) Install the preview2 SDK: https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.1.100-preview2-windows-x64-installer
2) Update dotnet ef tool:
```
PS C:\WINDOWS\system32> dotnet tool update dotnet-ef --global --version="3.1.0-preview2.19525.5"
Tool 'dotnet-ef' was successfully updated from version '3.0.0' to version '3.1.0-preview2.19525.5'.
```

```
PM> dotnet ef --version
Entity Framework Core .NET Command-line Tools
3.1.0-preview2.19525.5
```

3) dotnet ef migrations add / dotnet ef database update works now
